### PR TITLE
i#3544 RV64: Small improvments to some assembly functions

### PR DESCRIPTION
**Pull request by ksco**
This patch optimizes some basic assembly functions and adds some comments to improve readability.

Issue: #3544

### CODE REVIEW

File: core/arch/riscv64/riscv64.asm [hunk 1]
```diff
@@ -368,8 +368,7 @@ GLOBAL_LABEL(dr_try_start:)
         DECLARE_FUNC(dr_setjmp)
```
Comment (abhinav92003) [inline core/arch/riscv64/riscv64.asm:368]: Per the commit messages section at https://dynamorio.org/page_code_reviews.html: we prefer to not keep the PR description empty. Please add a description of the changes made here.

File: core/arch/riscv64/riscv64.asm [hunk 2]
```diff
@@ -395,11 +394,9 @@ GLOBAL_LABEL(dr_setjmp:)
         fsd      fs10, 24*ARG_SZ (ARG1)
         fsd      fs11, 25*ARG_SZ (ARG1)
 # ifdef UNIX
-        addi     sp, sp, -16
-        sd       ra, 0 (sp)
+        mv       t0, ra
```
Comment (abhinav92003) [inline core/arch/riscv64/riscv64.asm:397]: Based on the RISC-V calling convention, t0 is volatile across calls.

**Review by abhinav92003 — CHANGES_REQUESTED** [review]

File: core/arch/riscv64/riscv64.asm (hunk 1 above)
Comment (ksco) [reply core/arch/riscv64/riscv64.asm:368]: Done

File: core/arch/riscv64/riscv64.asm (hunk 2 above)
Comment (ksco) [reply core/arch/riscv64/riscv64.asm]: Sorry, I reverted this change.

**Review by abhinav92003 — APPROVED** [review]

**Pull request merged**